src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -719,37 +734,74 @@ class Wavefunction
         return kernels::subsytemwavefunction(wfn_, get_qubit_positions(qs), qubitswfn, tolerance);
     }
 
-    // apply permutation of basis states to the wave function
+    /// Apply the unitary operator that permutes the standard computational basis of the subsystem, defined by the
+    /// provided qubits. `qs` lists the qubits of the subsystem in little-endian order (the front qubit in the list
+    /// corresponds to the last position of the standard computational basis). `table_size` must be equal 2^n, where n
+    /// is the number of qubits in `qs` `permutation_table` must point to an array of `table_size` integers that
+    /// describes a permutation of set {0, 1, ... , 2^n -1}. `adjoint` specifies whether should apply the adjoint of the
+    /// unitary.
```

Comment:
(minor language nit)
```suggestion
    /// is the number of qubits in `qs`. `permutation_table` must point to an array of `table_size` integers that
    /// describes a permutation of set {0, 1, ... , 2^n -1}. `adjoint` specifies whether to apply the adjoint of the
    /// unitary.
```